.github/workflows/ci.yaml

```diff
@@ -38,7 +38,7 @@ env:
   MYPY_CACHE_VERSION: 7
   HA_SHORT_VERSION: "2024.4"
   DEFAULT_PYTHON: "3.11"
-  ALL_PYTHON_VERSIONS: "['3.11', '3.12']"
+  ALL_PYTHON_VERSIONS: "['3.12']"
```

Comment:
This will fail once the base env isn't cached anymore / the requirements change. The default version needs to be one of the `ALL_PYTHON_VERSIONS`.

The only change required is a simple mypy fix - adding another `type: ignore` here:
https://github.com/home-assistant/core/blob/f8b05a0a81b802dc91ec8da60d70d8b32cf03c7e/homeassistant/helpers/config_validation.py#L905-L908

```diff
                 near = (
                     f"near {config.__config_file__}"  # type: ignore[attr-defined]
-                    f":{config.__line__} "
+                    f":{config.__line__} "  # type: ignore[attr-defined]
                 )
```
That's the result of a slightly different AST for f-strings in Python 3.12 which is why I couldn't add that earlier. Formatting the line differently would have been over the line-length, see #108840.

Reply:
> `DEFAULT_PYTHON` is used by non-tests cases, which have their own cache keys.

`DEFAULT_PYTHON` is also used for `mypy`, `pylint`, `hassfest` and `gen-requirements-all`. They just use the full base env which is why this would fail once it changes.

> We still need to ship Python 3.11-supported patch releases, we must keep our tooling at Python 3.11, which is what this PR does.

If you want to keep the tooling at `3.11` for now, you'd need to change the pytest matrix values to only use `3.12` there.

Reply:
I see! Thanks! 👍